packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -308,7 +308,7 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
     }
 
     protected async doSave(reason: TextDocumentSaveReason, token: CancellationToken): Promise<void> {
-        if (token.isCancellationRequested || !this.resource.saveContents || !this.dirty) {
+        if (token.isCancellationRequested || !this.resource.saveContents) {
```

Comment:
I guess what @lmcbout wanted to do here is "formatting the file on users hitting CTRL+S even if the file is not dirty" - vsCode does it.

Reply:
This is exactly the reason, to allow "CTRL+S" like vscode does

Reply:
Please check whether VS Code actually save changes or only try to apply formatting. In our case sending unchanged content is expensive and we should avoid doing it.

Reply:
I tried  VSCode with the file (theia/.vscode/launch.json). The file is not dirty. By selecting "CTRL+S", it reformatted the file and saved it.

Reply:
I meant in our case if formatting does nothing we should not write anything to the disk. It is expensive for large files.

Reply:
After the formatting, the frontend is using the monaco-editor-model.doSave()
In that method, we check if there is some changes (~ line 320) code:
const changes = this.popContentChanges();
        if (changes.length === 0) {
            return;
        }
This is the same as testing isDirty(). In fact we could use:
if (this.dirty) return;
 to replace those lines above and it would do the same here . So as a resume, the file  remains in the frontend if there is no formatting , they are not send to the backend.